### PR TITLE
Add Keras ImageNet training example and LearningRateScheduleCallback

### PR DESCRIPTION
**Pull request by alsrgv**

### CODE REVIEW

File: examples/keras_imagenet_resnet50.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#
+# ResNet-50 model training using Keras and Horovod.
+#
+# This model is an example of computation-intensive model that achieves good accuracy on an image
```
Comment (sblotner) [inline examples/keras_imagenet_resnet50.py]: of computation-intensive --> of a computation-intensive

File: examples/keras_imagenet_resnet50.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+# classification task.  It brings together distributed training concepts such as learning rate
+# schedule adjustments with a warmup, randomized data reading and checkpointing on the first worker
```
Comment (sblotner) [inline examples/keras_imagenet_resnet50.py]: add comma between reading + and

File: examples/keras_imagenet_resnet50.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+# only.
+#
+# Note: this model uses Keras native ImageDataGenerator and not the sophisticated preprocessing
```
Comment (sblotner) [inline examples/keras_imagenet_resnet50.py]: Capitalize "this"

File: examples/keras_imagenet_resnet50.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+# pipeline that is typically used to train state-of-the-art ResNet-50 model.  This results in ~0.5%
+# increase in top-1 validation error compared to https://github.com/KaimingHe/deep-residual-networks.
```
Comment (sblotner) [inline examples/keras_imagenet_resnet50.py]: in top-1 --> in the top-1

File: examples/keras_imagenet_resnet50.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+callbacks = [
+    # Horovod: broadcast initial variable states from rank 0 to all other processes.
+    # This is necessary to ensure consistent initialization of all workers when
+    # training is started with random weights or restored from a checkpoint.
+    hvd.callbacks.BroadcastGlobalVariablesCallback(0),
+
+    # Horovod: average metrics among workers at the end of every epoch.
+    #
+    # Note: This callback must be in the list before the ReduceLROnPlateau,
+    # TensorBoard or other metrics-based callbacks.
```
Comment (sblotner) [inline examples/keras_imagenet_resnet50.py]: add comma after "TensorBoard"

File: horovod/keras/callbacks.py [hunk 6]
```diff
@@ -157,31 +179,79 @@ def on_epoch_begin(self, epoch, logs=None):
...
+        lr'(epoch = 0)      = ----
+                              size
+
+        lr'(epoch = warmup) = lr
+    """
+
+    def __init__(self, warmup_epochs=5, momentum_correction=True, steps_per_epoch=None,
+                 verbose=0):
+        """
+        Construct a new LearningRateWarmupCallback that will gradually warmup learning rate.
```
Comment (sblotner) [inline horovod/keras/callbacks.py]: warmup --> warm up the

File: horovod/keras/callbacks.py [hunk 7]
```diff
@@ -145,43 +150,110 @@ def _autodetect_steps_per_epoch(self):
...
             self.steps_per_epoch = self._autodetect_steps_per_epoch()
 
     def on_epoch_begin(self, epoch, logs=None):
         self.current_epoch = epoch
 
     def on_batch_begin(self, batch, logs=None):
-        if self.current_epoch > self.warmup_epochs:
-            # Outside of adjustment scope.
+        if (self.current_epoch < self.start_epoch or
+                self.end_epoch is not None and self.current_epoch >= self.end_epoch):
```
Comment (jeevandev) [inline horovod/keras/callbacks.py]: Pls enclose the and condition with parentheses so that no need to worry about operator precedence

**Pull request merged**